### PR TITLE
refactor(cohort-filtering): replace cohort drawer with detail page

### PR DESCRIPTION
**Pull request by alexkim205**
## Problem

As part of the new cohort filtering experience, we're moving away from cohort drawers and towards individual detail pages for each cohort.

Note: this PR does not implement any of the new mockup changes regarding filters, matching groups, sections, etc. It solely rewires routing logic and keeps the existing experience intact (bugs and all). 

## Changes

https://www.figma.com/file/gQBj9YnNgD8YW4nBwCVLZf/PostHog-App?node-id=8919%3A55628

**Old behavior**

https://user-images.githubusercontent.com/13460330/163032972-ae9b6697-1af9-4cc2-b67c-99c39972c5cd.mov

**New behavior**

https://user-images.githubusercontent.com/13460330/163032896-14374c4d-1c70-4c4a-90ec-00578e03cb91.mov

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

## How did you test this code?

Existing e2e tests don't break. 

Creating, updating, and deleting cohorts still work.

### CODE REVIEW

File: frontend/src/types.ts [hunk 1]
```diff
@@ -577,7 +577,7 @@ export interface CohortType {
     created_by?: UserBasicType | null
     created_at?: string
     deleted?: boolean
-    id: number | 'new' | 'personsModalNew'
```
Comment (alexkim205) [inline frontend/src/types.ts:580]: `personsModalNew` not used anywhere

**Review by liyiy — DISMISSED** [review]
Creating a new cohort is not working correctly on this, it loads forever but on master it doesn't do the same

Screen Shot 2022-04-13 at 10 06 22 AM

**Comment by alexkim205** [comment]
> Creating a new cohort is not working correctly on this, it loads forever but on master it doesn't do the same

Fixed!

**Review by clarkus — APPROVED** [review]

**Review by liyiy — APPROVED** [review]
Seems fine and not broken

**Review by EDsCODE — APPROVED** [review]

**Pull request merged**